### PR TITLE
Allow for '-o feature@<feature>=disabled' on the command line.

### PR DESCRIPTION
**Pull request by loli10K**
Based on @FransUrbo original work from previous PR https://github.com/zfsonlinux/zfs/pull/3465.

Enables new test script `zpool_create_features_005_pos` in Linux runfile and updates existing `zpool_create_features_004_neg` test case.

In case of invalid property value libzfs still emits a misleading error "_cannot create 'tank': property 'feature@async_destroy' can only be set to 'enabled'_" even if now _'disabled'_ is also accepted ...

Should fix https://github.com/zfsonlinux/zfs/issues/3460.

### CODE REVIEW

File: tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_features_005_pos.ksh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		# Feature is not set - failure.
+		return 1
+	fi
+}
+
+log_onexit cleanup
+
+for feature in async_destroy bookmarks embedded_data empty_bpobj enabled_txg \
+               extensible_dataset filesystem_limits hole_birth large_blocks  \
+               lz4_compress spacemap_histogram
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_features_005_pos.ksh]: We're the large_dnode and userobj_accounting intentionally skipped?

File: man/man8/zpool.8 [hunk 2]
```diff
@@ -41,8 +41,9 @@ zpool \- configures ZFS storage pools
 
 .LP
 .nf
-\fBzpool create\fR [\fB-fnd\fR] [\fB-o\fR \fIproperty=value\fR] ... [\fB-O\fR \fIfile-system-property=value\fR]
-     ... [\fB-m\fR \fImountpoint\fR] [\fB-R\fR \fIroot\fR] [\fB-t\fR \fItname\fR] \fIpool\fR \fIvdev\fR ...
+\fBzpool create\fR [\fB-fnd\fR] [\fB-o\fR \fIproperty=value\fR] ... [\fB-o\fR feature@\fIfeature=value\fR]
+     ... [\fB-O\fR \fIfile-system-property=value\fR] ... [\fB-m\fR \fImountpoint\fR] [\fB-R\fR \fIroot\fR] [\fB-t\fR \fItname\fR]
+     ... \fIpool\fR \fIvdev\fR ...
```
Comment (behlendorf) [inline man/man8/zpool.8]: Can you fix the line wrapping here.  [-t name] needs to be moved to the last line with ... pool vdev ...

```
       zpool create [-fnd] [-o property=value] ... [-o feature@feature=value]
            ... [-O file-system-property=value] ... [-m mountpoint] [-R root]
[-t tname]
            ... pool vdev ...
```

**Comment by behlendorf** [comment]
Succeeds #3465.  @FransUrbo, @gordan-bobic, @DeHackEd if you get a chance can you please review and test this.

**Comment by loli10K** [comment]
> Is there some reason you didn't take this in the original patch?

I just wanted to stay as close as possible to the original PR, i will update that error message in libzfs along with the other stuff.

**Comment by behlendorf** [comment]
@loli10K yes, we might as well add the full set of supported features.

**Comment by loli10K** [comment]
Changes up. Also i just realized this should probably fix https://github.com/zfsonlinux/zfs/issues/5142.

**Comment by behlendorf** [comment]
@loli10K the patch itself looks good but the test case wasn't working as you originally intended.  I've added commit b06b75e to this branch which addresses the issues with the test case.  Please review those changes.

**Comment by loli10K** [comment]
LGTM, but now i wonder if we should also test multiple `-o feature@<feature>=disabled|enabled` with various permutations to exercise the cli parsing code of this patch?

**Comment by behlendorf** [comment]
@loli10K that would be a nice addition but I don't think it's critical.  The parsing logic is pretty straight forward and I did manually run some of those kind of tests and it did work as expected.  Thanks for the quick review.

**Comment by behlendorf** [comment]
@loli10K thanks for tackling this, I know this is something quite a few people were requesting.

**Comment by loli10K** [comment]
@behlendorf that's the reason i did it, i saw different issues being open for this and i thought someone had to implement it.

I know i'm being offtopic here, but do you happen to know if some other developer is working on recv property overrides (-x|-o)?

**Comment by behlendorf** [comment]
@loli10K I don't know of anyone working on the recv property overrides.  There was a WIP patch in 1867 along with a first round of review comments.

**Comment by ahrens** [comment]
On the recv property overrides, IIRC my comments were essentially, "why are the semantics subtly different from those of Oracle Solaris"?

**Comment by loli10K** [comment]
@ahrens yes, that's exactly as i read them. Is anyone in the Illumos community working on this? The open issue here https://www.illumos.org/issues/2745 doesn't show much activity either ...

**Comment by ryao** [comment]
@loli10K I can confirm @behlendorf's remark about not knowing anyone working on recv property overrides in the Linux community. I don't know of anyone either.

**Pull request merged**